source/ui/CLS/CLSSIS3820ScalerControlsView.cpp

```diff
@@ -134,102 +127,153 @@ void CLSSIS3820ScalerControlsView::stopScanning()
 
 void CLSSIS3820ScalerControlsView::setContinuous()
 {
-	if (scaler_) {
-		int index = modeChoice_->currentIndex();
-
-		if (index == 1 && !scaler_->isContinuous())
-			scaler_->setContinuous(true);
-
-		else if (index == 0 && scaler_->isContinuous())
-			scaler_->setContinuous(false);
+	if (scaler_ && scaler_->isConnected()) {
+		bool modeOK = false;
+		int mode = acquisitionModeBox_->itemData(acquisitionModeBox_->currentIndex()).toInt(&modeOK);
+
+		if (modeOK) {
+			if (mode == CLSSIS3820Scaler::Continuous)
+				scaler_->setContinuous(true);
+			else if (mode == CLSSIS3820Scaler::SingleShot)
+				scaler_->setContinuous(false);
+		}
 	}
 }
 
 void CLSSIS3820ScalerControlsView::setDwellTime()
 {
-	if (scaler_) {
-		double secondsDwell = (double)time_->value() / MILLISECONDS_PER_SECOND;
-
-		if (secondsDwell != scaler_->dwellTime())
-			scaler_->setDwellTime(secondsDwell);
+	if (scaler_ && scaler_->isConnected()) {
+		double dwellTime = dwellTimeBox_->value() / 1000.0; // The dwell time box displays time in ms, the scaler keeps time in s.
+		scaler_->setDwellTime(dwellTime);
 	}
 }
 
 void CLSSIS3820ScalerControlsView::setScansPerBuffer()
 {
-	if (scaler_) {
-		int newValue = scansPerBuffer_->value();
-
-		if (newValue != scaler_->scansPerBuffer())
-			scaler_->setScansPerBuffer(newValue);
+	if (scaler_ && scaler_->isConnected()) {
+		int newValue = scansPerBufferBox_->value();
```

Comment:
I understand the desire to have the dwell time be on separate lines because of the operation you do to it, but this one and the next ones really should be a single call.